### ISSUE TITLE
[Docker container]  DL framework (tf) reports GPU acceleration is disabled. This could be due to a bad CUDA- or tf installation.

### ISSUE THREAD

**Issue by cambel** [ray-project/ray#14560]
### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):*
Python: Python 3.7.7
Tensorflow: 2.4.1
OS: Ubuntu 18.04

Similarly to [#13735](https://github.com/ray-project/ray/issues/13735) RAY does not recognize the GPU:

> GPUs were assigned to this worker by Ray, but your DL framework (tf) reports GPU acceleration is disabled. This could be due to a bad CUDA- or tf installation.

However, Tensorflow does recognize the GPU device:
`python -c "import tensorflow as tf; tf.config.list_physical_devices()"`

> 2021-03-09 00:21:26.385517: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: 
> pciBusID: 0000:01:00.0 name: GeForce RTX 2060 SUPER computeCapability: 7.5
> coreClock: 1.65GHz coreCount: 34 deviceMemorySize: 7.79GiB deviceMemoryBandwidth: 417.29GiB/s

I am testing the basic example provided in the [README](https://github.com/ray-project/ray#rllib-quick-start)
I compile the [dockerfile](https://github.com/ray-project/ray/blob/master/docker/ray-ml/Dockerfile) for ML 

Here is the full log when I execute the example script

> 3-09 00:15:15,982	INFO services.py:1251 -- View the Ray dashboard at http://127.0.0.1:8265
> WARNING:tensorflow:From /home/ray/anaconda3/lib/python3.7/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
> Instructions for updating:
> non-resource variables are not supported in the long term
> Loading environment football failed: No module named 'gfootball'
> == Status ==
> Memory usage on this node: 9.3/31.2 GiB
> Using FIFO scheduling algorithm.
> Resources requested: 0/16 CPUs, 0/1 GPUs, 0.0/25.02 GiB heap, 0.0/12.51 GiB objects (0.0/1.0 accelerator_type:RTX)
> Result logdir: /home/ray/ray_results/PPO
> Number of trials: 1/1 (1 PENDING)
> +--------------------------------+----------+-------+
> | Trial name                     | status   | loc   |
> |--------------------------------+----------+-------|
> | PPO_SimpleCorridor_96575_00000 | PENDING  |       |
> +--------------------------------+----------+-------+
> 
> 
> (pid=197) WARNING:tensorflow:From /home/ray/anaconda3/lib/python3.7/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
> (pid=197) Instructions for updating:
> (pid=197) non-resource variables are not supported in the long term
> (pid=197) Loading environment football failed: No module named 'gfootball'
> (pid=197) 2021-03-09 00:15:21,666	INFO trainer.py:660 -- Tip: set framework=tfe or the --eager flag to enable TensorFlow eager execution
> (pid=195) WARNING:tensorflow:From /home/ray/anaconda3/lib/python3.7/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
> (pid=195) Instructions for updating:
> (pid=195) non-resource variables are not supported in the long term
> (pid=185) WARNING:tensorflow:From /home/ray/anaconda3/lib/python3.7/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
> (pid=185) Instructions for updating:
> (pid=185) non-resource variables are not supported in the long term
> (pid=188) WARNING:tensorflow:From /home/ray/anaconda3/lib/python3.7/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
> (pid=188) Instructions for updating:
> (pid=188) non-resource variables are not supported in the long term
> (pid=196) WARNING:tensorflow:From /home/ray/anaconda3/lib/python3.7/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
> (pid=196) Instructions for updating:
> (pid=196) non-resource variables are not supported in the long term
> (pid=188) Loading environment football failed: No module named 'gfootball'
> (pid=185) Loading environment football failed: No module named 'gfootball'
> (pid=195) Loading environment football failed: No module named 'gfootball'
> (pid=196) Loading environment football failed: No module named 'gfootball'
> (pid=188) 2021-03-09 00:15:23,915	DEBUG worker_set.py:253 -- Creating TF session {'intra_op_parallelism_threads': 2, 'inter_op_parallelism_threads': 2, 'gpu_options': {'allow_growth': True}, 'log_device_placement': False, 'device_count': {'CPU': 1}, 'allow_soft_placement': True}
> (pid=195) 2021-03-09 00:15:23,941	DEBUG worker_set.py:253 -- Creating TF session {'intra_op_parallelism_threads': 2, 'inter_op_parallelism_threads': 2, 'gpu_options': {'allow_growth': True}, 'log_device_placement': False, 'device_count': {'CPU': 1}, 'allow_soft_placement': True}
> (pid=195) 2021-03-09 00:15:23,945	DEBUG rollout_worker.py:1093 -- Creating policy for default_policy
> (pid=195) 2021-03-09 00:15:23,946	DEBUG catalog.py:632 -- Created preprocessor <ray.rllib.models.preprocessors.NoPreprocessor object at 0x7efd98269ad0>: Box(0.0, 5.0, (1,), float32) -> (1,)
> (pid=185) 2021-03-09 00:15:23,929	DEBUG worker_set.py:253 -- Creating TF session {'intra_op_parallelism_threads': 2, 'inter_op_parallelism_threads': 2, 'gpu_options': {'allow_growth': True}, 'log_device_placement': False, 'device_count': {'CPU': 1}, 'allow_soft_placement': True}
> (pid=185) 2021-03-09 00:15:23,959	DEBUG rollout_worker.py:1093 -- Creating policy for default_policy
> (pid=185) 2021-03-09 00:15:23,959	DEBUG catalog.py:632 -- Created preprocessor <ray.rllib.models.preprocessors.NoPreprocessor object at 0x7ff2ba0b1b10>: Box(0.0, 5.0, (1,), float32) -> (1,)
> (pid=188) 2021-03-09 00:15:23,948	DEBUG rollout_worker.py:1093 -- Creating policy for default_policy
> (pid=188) 2021-03-09 00:15:23,948	DEBUG catalog.py:632 -- Created preprocessor <ray.rllib.models.preprocessors.NoPreprocessor object at 0x7f1b2e2b1ad0>: Box(0.0, 5.0, (1,), float32) -> (1,)
> (pid=196) 2021-03-09 00:15:23,943	DEBUG worker_set.py:253 -- Creating TF session {'intra_op_parallelism_threads': 2, 'inter_op_parallelism_threads': 2, 'gpu_options': {'allow_growth': True}, 'log_device_placement': False, 'device_count': {'CPU': 1}, 'allow_soft_placement': True}
> (pid=196) 2021-03-09 00:15:23,948	DEBUG rollout_worker.py:1093 -- Creating policy for default_policy
> (pid=196) 2021-03-09 00:15:23,948	DEBUG catalog.py:632 -- Created preprocessor <ray.rllib.models.preprocessors.NoPreprocessor object at 0x7ef892df2b10>: Box(0.0, 5.0, (1,), float32) -> (1,)
> (pid=195) 2021-03-09 00:15:24,068	INFO dynamic_tf_policy.py:482 -- Testing `compute_actions` w/ dummy batch.
> (pid=188) 2021-03-09 00:15:24,068	INFO dynamic_tf_policy.py:482 -- Testing `compute_actions` w/ dummy batch.
> (pid=196) 2021-03-09 00:15:24,069	INFO dynamic_tf_policy.py:482 -- Testing `compute_actions` w/ dummy batch.
> (pid=196) 2021-03-09 00:15:24,070	INFO tf_run_builder.py:87 -- Executing TF run without tracing. To dump TF timeline traces to disk, set the TF_TIMELINE_DIR environment variable.
> (pid=195) 2021-03-09 00:15:24,101	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_prob` to view-reqs.
> (pid=195) 2021-03-09 00:15:24,102	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_logp` to view-reqs.
> (pid=195) 2021-03-09 00:15:24,102	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_dist_inputs` to view-reqs.
> (pid=195) 2021-03-09 00:15:24,103	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `vf_preds` to view-reqs.
> (pid=195) 2021-03-09 00:15:24,103	INFO dynamic_tf_policy.py:533 -- Testing `postprocess_trajectory` w/ dummy batch.
> (pid=185) 2021-03-09 00:15:24,081	INFO dynamic_tf_policy.py:482 -- Testing `compute_actions` w/ dummy batch.
> (pid=185) 2021-03-09 00:15:24,116	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_prob` to view-reqs.
> (pid=185) 2021-03-09 00:15:24,117	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_logp` to view-reqs.
> (pid=185) 2021-03-09 00:15:24,117	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_dist_inputs` to view-reqs.
> (pid=185) 2021-03-09 00:15:24,117	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `vf_preds` to view-reqs.
> (pid=185) 2021-03-09 00:15:24,117	INFO dynamic_tf_policy.py:533 -- Testing `postprocess_trajectory` w/ dummy batch.
> (pid=188) 2021-03-09 00:15:24,101	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_prob` to view-reqs.
> (pid=188) 2021-03-09 00:15:24,101	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_logp` to view-reqs.
> (pid=188) 2021-03-09 00:15:24,102	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_dist_inputs` to view-reqs.
> (pid=188) 2021-03-09 00:15:24,102	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `vf_preds` to view-reqs.
> (pid=188) 2021-03-09 00:15:24,102	INFO dynamic_tf_policy.py:533 -- Testing `postprocess_trajectory` w/ dummy batch.
> (pid=196) 2021-03-09 00:15:24,103	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_prob` to view-reqs.
> (pid=196) 2021-03-09 00:15:24,103	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_logp` to view-reqs.
> (pid=196) 2021-03-09 00:15:24,103	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_dist_inputs` to view-reqs.
> (pid=196) 2021-03-09 00:15:24,104	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `vf_preds` to view-reqs.
> (pid=196) 2021-03-09 00:15:24,104	INFO dynamic_tf_policy.py:533 -- Testing `postprocess_trajectory` w/ dummy batch.
> (pid=196) 2021-03-09 00:15:24,125	DEBUG dynamic_tf_policy.py:585 -- Initializing loss function with dummy input:
> (pid=196) 
> (pid=196) { 'action_dist_inputs': <tf.Tensor 'default_policy/action_dist_inputs:0' shape=(?, 2) dtype=float32>,
> (pid=196)   'action_logp': <tf.Tensor 'default_policy/action_logp:0' shape=(?,) dtype=float32>,
> (pid=196)   'action_prob': <tf.Tensor 'default_policy/action_prob:0' shape=(?,) dtype=float32>,
> (pid=196)   'actions': <tf.Tensor 'default_policy/action:0' shape=(?,) dtype=int64>,
> (pid=196)   'advantages': <tf.Tensor 'default_policy/advantages:0' shape=(?,) dtype=float32>,
> (pid=196)   'agent_index': <tf.Tensor 'default_policy/agent_index:0' shape=(?,) dtype=float32>,
> (pid=196)   'dones': <tf.Tensor 'default_policy/dones:0' shape=(?,) dtype=float32>,
> (pid=196)   'eps_id': <tf.Tensor 'default_policy/eps_id:0' shape=(?,) dtype=float32>,
> (pid=196)   'infos': <tf.Tensor 'default_policy/infos:0' shape=(?,) dtype=float32>,
> (pid=196)   'is_training': <tf.Tensor 'default_policy/is_training:0' shape=() dtype=bool>,
> (pid=196)   'new_obs': <tf.Tensor 'default_policy/new_obs:0' shape=(?, 1) dtype=float32>,
> (pid=196)   'obs': <tf.Tensor 'default_policy/obs:0' shape=(?, 1) dtype=float32>,
> (pid=196)   'prev_actions': <tf.Tensor 'default_policy/prev_action:0' shape=(?,) dtype=int64>,
> (pid=196)   'prev_rewards': <tf.Tensor 'default_policy/prev_rewards:0' shape=(?,) dtype=float32>,
> (pid=196)   'rewards': <tf.Tensor 'default_policy/rewards:0' shape=(?,) dtype=float32>,
> (pid=196)   't': <tf.Tensor 'default_policy/t:0' shape=(?,) dtype=float32>,
> (pid=196)   'unroll_id': <tf.Tensor 'default_policy/unroll_id:0' shape=(?,) dtype=float32>,
> (pid=196)   'value_targets': <tf.Tensor 'default_policy/value_targets:0' shape=(?,) dtype=float32>,
> (pid=196)   'vf_preds': <tf.Tensor 'default_policy/vf_preds:0' shape=(?,) dtype=float32>}
> (pid=196) 
> (pid=196) 2021-03-09 00:15:24,408	DEBUG tf_policy.py:304 -- These tensors were used in the loss_fn:
> (pid=196) 
> (pid=196) { 'action_dist_inputs': <tf.Tensor 'default_policy/action_dist_inputs:0' shape=(?, 2) dtype=float32>,
> (pid=196)   'action_logp': <tf.Tensor 'default_policy/action_logp:0' shape=(?,) dtype=float32>,
> (pid=196)   'actions': <tf.Tensor 'default_policy/action:0' shape=(?,) dtype=int64>,
> (pid=196)   'advantages': <tf.Tensor 'default_policy/advantages:0' shape=(?,) dtype=float32>,
> (pid=196)   'dones': <tf.Tensor 'default_policy/dones:0' shape=(?,) dtype=float32>,
> (pid=196)   'new_obs': <tf.Tensor 'default_policy/new_obs:0' shape=(?, 1) dtype=float32>,
> (pid=196)   'obs': <tf.Tensor 'default_policy/obs:0' shape=(?, 1) dtype=float32>,
> (pid=196)   'rewards': <tf.Tensor 'default_policy/rewards:0' shape=(?,) dtype=float32>,
> (pid=196)   'value_targets': <tf.Tensor 'default_policy/value_targets:0' shape=(?,) dtype=float32>,
> (pid=196)   'vf_preds': <tf.Tensor 'default_policy/vf_preds:0' shape=(?,) dtype=float32>}
> (pid=196) 
> (pid=197) 2021-03-09 00:15:24,520	DEBUG worker_set.py:253 -- Creating TF session {'intra_op_parallelism_threads': 8, 'inter_op_parallelism_threads': 8, 'gpu_options': {'allow_growth': True}, 'log_device_placement': False, 'device_count': {'CPU': 1}, 'allow_soft_placement': True}
> (pid=197) 2021-03-09 00:15:24,525	DEBUG rollout_worker.py:1093 -- Creating policy for default_policy
> (pid=197) 2021-03-09 00:15:24,525	DEBUG catalog.py:632 -- Created preprocessor <ray.rllib.models.preprocessors.NoPreprocessor object at 0x7f13c9577390>: Box(0.0, 5.0, (1,), float32) -> (1,)
> (pid=195) 2021-03-09 00:15:24,517	DEBUG rollout_worker.py:503 -- Creating policy evaluation worker 2 on CPU (please ignore any CUDA init errors)
> (pid=195) 2021-03-09 00:15:24,518	DEBUG rollout_worker.py:652 -- Created rollout worker with env <ray.rllib.env.base_env._VectorEnvToBaseEnv object at 0x7efd98269990> (<SimpleCorridor instance>), policies {'default_policy': <ray.rllib.policy.tf_policy_template.PPOTFPolicy object at 0x7efd98269e90>}
> (pid=188) 2021-03-09 00:15:24,506	DEBUG rollout_worker.py:503 -- Creating policy evaluation worker 3 on CPU (please ignore any CUDA init errors)
> (pid=188) 2021-03-09 00:15:24,507	DEBUG rollout_worker.py:652 -- Created rollout worker with env <ray.rllib.env.base_env._VectorEnvToBaseEnv object at 0x7f1b2c13cf50> (<SimpleCorridor instance>), policies {'default_policy': <ray.rllib.policy.tf_policy_template.PPOTFPolicy object at 0x7f1b2e2b1e10>}
> (pid=196) 2021-03-09 00:15:24,517	DEBUG rollout_worker.py:503 -- Creating policy evaluation worker 1 on CPU (please ignore any CUDA init errors)
> (pid=196) 2021-03-09 00:15:24,517	DEBUG rollout_worker.py:652 -- Created rollout worker with env <ray.rllib.env.base_env._VectorEnvToBaseEnv object at 0x7ef892df2a10> (<SimpleCorridor instance>), policies {'default_policy': <ray.rllib.policy.tf_policy_template.PPOTFPolicy object at 0x7ef892df2e90>}
> (pid=185) 2021-03-09 00:15:24,539	DEBUG rollout_worker.py:503 -- Creating policy evaluation worker 4 on CPU (please ignore any CUDA init errors)
> (pid=185) 2021-03-09 00:15:24,539	DEBUG rollout_worker.py:652 -- Created rollout worker with env <ray.rllib.env.base_env._VectorEnvToBaseEnv object at 0x7ff2ba0b1a10> (<SimpleCorridor instance>), policies {'default_policy': <ray.rllib.policy.tf_policy_template.PPOTFPolicy object at 0x7ff2ba0b1e90>}
> (pid=197) 2021-03-09 00:15:24,642	INFO dynamic_tf_policy.py:482 -- Testing `compute_actions` w/ dummy batch.
> (pid=197) 2021-03-09 00:15:24,643	INFO tf_run_builder.py:87 -- Executing TF run without tracing. To dump TF timeline traces to disk, set the TF_TIMELINE_DIR environment variable.
> (pid=197) 2021-03-09 00:15:24,674	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_prob` to view-reqs.
> (pid=197) 2021-03-09 00:15:24,674	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_logp` to view-reqs.
> (pid=197) 2021-03-09 00:15:24,675	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `action_dist_inputs` to view-reqs.
> (pid=197) 2021-03-09 00:15:24,675	INFO dynamic_tf_policy.py:491 -- Adding extra-action-fetch `vf_preds` to view-reqs.
> (pid=197) 2021-03-09 00:15:24,675	INFO dynamic_tf_policy.py:533 -- Testing `postprocess_trajectory` w/ dummy batch.
> (pid=197) 2021-03-09 00:15:24,693	DEBUG dynamic_tf_policy.py:585 -- Initializing loss function with dummy input:
> (pid=197) 
> (pid=197) { 'action_dist_inputs': <tf.Tensor 'default_policy/action_dist_inputs:0' shape=(?, 2) dtype=float32>,
> (pid=197)   'action_logp': <tf.Tensor 'default_policy/action_logp:0' shape=(?,) dtype=float32>,
> (pid=197)   'action_prob': <tf.Tensor 'default_policy/action_prob:0' shape=(?,) dtype=float32>,
> (pid=197)   'actions': <tf.Tensor 'default_policy/action:0' shape=(?,) dtype=int64>,
> (pid=197)   'advantages': <tf.Tensor 'default_policy/advantages:0' shape=(?,) dtype=float32>,
> (pid=197)   'agent_index': <tf.Tensor 'default_policy/agent_index:0' shape=(?,) dtype=float32>,
> (pid=197)   'dones': <tf.Tensor 'default_policy/dones:0' shape=(?,) dtype=float32>,
> (pid=197)   'eps_id': <tf.Tensor 'default_policy/eps_id:0' shape=(?,) dtype=float32>,
> (pid=197)   'infos': <tf.Tensor 'default_policy/infos:0' shape=(?,) dtype=float32>,
> (pid=197)   'is_training': <tf.Tensor 'default_policy/is_training:0' shape=() dtype=bool>,
> (pid=197)   'new_obs': <tf.Tensor 'default_policy/new_obs:0' shape=(?, 1) dtype=float32>,
> (pid=197)   'obs': <tf.Tensor 'default_policy/obs:0' shape=(?, 1) dtype=float32>,
> (pid=197)   'prev_actions': <tf.Tensor 'default_policy/prev_action:0' shape=(?,) dtype=int64>,
> (pid=197)   'prev_rewards': <tf.Tensor 'default_policy/prev_rewards:0' shape=(?,) dtype=float32>,
> (pid=197)   'rewards': <tf.Tensor 'default_policy/rewards:0' shape=(?,) dtype=float32>,
> (pid=197)   't': <tf.Tensor 'default_policy/t:0' shape=(?,) dtype=float32>,
> (pid=197)   'unroll_id': <tf.Tensor 'default_policy/unroll_id:0' shape=(?,) dtype=float32>,
> (pid=197)   'value_targets': <tf.Tensor 'default_policy/value_targets:0' shape=(?,) dtype=float32>,
> (pid=197)   'vf_preds': <tf.Tensor 'default_policy/vf_preds:0' shape=(?,) dtype=float32>}
> (pid=197) 
> (pid=197) 2021-03-09 00:15:24,956	DEBUG tf_policy.py:304 -- These tensors were used in the loss_fn:
> (pid=197) 
> (pid=197) { 'action_dist_inputs': <tf.Tensor 'default_policy/action_dist_inputs:0' shape=(?, 2) dtype=float32>,
> (pid=197)   'action_logp': <tf.Tensor 'default_policy/action_logp:0' shape=(?,) dtype=float32>,
> (pid=197)   'actions': <tf.Tensor 'default_policy/action:0' shape=(?,) dtype=int64>,
> (pid=197)   'advantages': <tf.Tensor 'default_policy/advantages:0' shape=(?,) dtype=float32>,
> (pid=197)   'dones': <tf.Tensor 'default_policy/dones:0' shape=(?,) dtype=float32>,
> (pid=197)   'new_obs': <tf.Tensor 'default_policy/new_obs:0' shape=(?, 1) dtype=float32>,
> (pid=197)   'obs': <tf.Tensor 'default_policy/obs:0' shape=(?, 1) dtype=float32>,
> (pid=197)   'rewards': <tf.Tensor 'default_policy/rewards:0' shape=(?,) dtype=float32>,
> (pid=197)   'value_targets': <tf.Tensor 'default_policy/value_targets:0' shape=(?,) dtype=float32>,
> (pid=197)   'vf_preds': <tf.Tensor 'default_policy/vf_preds:0' shape=(?,) dtype=float32>}
> (pid=197) 
> 2021-03-09 00:15:25,060	ERROR trial_runner.py:727 -- Trial PPO_SimpleCorridor_96575_00000: Error processing event.
> Traceback (most recent call last):
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/trial_runner.py", line 697, in _process_trial
>     results = self.trial_executor.fetch_result(trial)
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/ray_trial_executor.py", line 678, in fetch_result
>     result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/client_mode_hook.py", line 47, in wrapper
>     return func(*args, **kwargs)
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/worker.py", line 1428, in get
>     raise value.as_instanceof_cause()
> ray.exceptions.RayTaskError(RuntimeError): ray::PPO.train_buffered() (pid=197, ip=10.0.1.150)
>   File "python/ray/_raylet.pyx", line 451, in ray._raylet.execute_task
>   File "python/ray/_raylet.pyx", line 484, in ray._raylet.execute_task
>   File "python/ray/_raylet.pyx", line 491, in ray._raylet.execute_task
>   File "python/ray/_raylet.pyx", line 501, in ray._raylet.execute_task
>   File "python/ray/_raylet.pyx", line 444, in ray._raylet.execute_task.function_executor
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/rllib/agents/trainer_template.py", line 121, in __init__
>     Trainer.__init__(self, config, env, logger_creator)
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/rllib/agents/trainer.py", line 513, in __init__
>     super().__init__(config, logger_creator)
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/trainable.py", line 98, in __init__
>     self.setup(copy.deepcopy(self.config))
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/rllib/agents/trainer.py", line 698, in setup
>     self._init(self.config, self.env_creator)
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/rllib/agents/trainer_template.py", line 153, in _init
>     num_workers=self.config["num_workers"])
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/rllib/agents/trainer.py", line 774, in _make_workers
>     logdir=self.logdir)
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/rllib/evaluation/worker_set.py", line 98, in __init__
>     spaces=spaces,
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/rllib/evaluation/worker_set.py", line 358, in _make_worker
>     _use_trajectory_view_api=config["_use_trajectory_view_api"],
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/rllib/evaluation/rollout_worker.py", line 513, in __init__
>     policy_config["framework"]))
> RuntimeError: GPUs were assigned to this worker by Ray, but your DL framework (tf) reports GPU acceleration is disabled. This could be due to a bad CUDA- or tf installation.
> Result for PPO_SimpleCorridor_96575_00000:
>   {}
>   
> == Status ==
> Memory usage on this node: 10.6/31.2 GiB
> Using FIFO scheduling algorithm.
> Resources requested: 0/16 CPUs, 0/1 GPUs, 0.0/25.02 GiB heap, 0.0/12.51 GiB objects (0.0/1.0 accelerator_type:RTX)
> Result logdir: /home/ray/ray_results/PPO
> Number of trials: 1/1 (1 ERROR)
> +--------------------------------+----------+-------+
> | Trial name                     | status   | loc   |
> |--------------------------------+----------+-------|
> | PPO_SimpleCorridor_96575_00000 | ERROR    |       |
> +--------------------------------+----------+-------+
> Number of errored trials: 1
> +--------------------------------+--------------+------------------------------------------------------------------------------------------+
> | Trial name                     |   # failures | error file                                                                               |
> |--------------------------------+--------------+------------------------------------------------------------------------------------------|
> | PPO_SimpleCorridor_96575_00000 |            1 | /home/ray/ray_results/PPO/PPO_SimpleCorridor_96575_00000_0_2021-03-09_00-15-19/error.txt |
> +--------------------------------+--------------+------------------------------------------------------------------------------------------+
> 
> == Status ==
> Memory usage on this node: 10.6/31.2 GiB
> Using FIFO scheduling algorithm.
> Resources requested: 0/16 CPUs, 0/1 GPUs, 0.0/25.02 GiB heap, 0.0/12.51 GiB objects (0.0/1.0 accelerator_type:RTX)
> Result logdir: /home/ray/ray_results/PPO
> Number of trials: 1/1 (1 ERROR)
> +--------------------------------+----------+-------+
> | Trial name                     | status   | loc   |
> |--------------------------------+----------+-------|
> | PPO_SimpleCorridor_96575_00000 | ERROR    |       |
> +--------------------------------+----------+-------+
> Number of errored trials: 1
> +--------------------------------+--------------+------------------------------------------------------------------------------------------+
> | Trial name                     |   # failures | error file                                                                               |
> |--------------------------------+--------------+------------------------------------------------------------------------------------------|
> | PPO_SimpleCorridor_96575_00000 |            1 | /home/ray/ray_results/PPO/PPO_SimpleCorridor_96575_00000_0_2021-03-09_00-15-19/error.txt |
> +--------------------------------+--------------+------------------------------------------------------------------------------------------+
> 
> (pid=197) 2021-03-09 00:15:25,056	INFO rollout_worker.py:1132 -- Built policy map: {'default_policy': <ray.rllib.policy.tf_policy_template.PPOTFPolicy object at 0x7f13c9577790>}
> (pid=197) 2021-03-09 00:15:25,056	INFO rollout_worker.py:1133 -- Built preprocessor map: {'default_policy': <ray.rllib.models.preprocessors.NoPreprocessor object at 0x7f13c9577390>}
> Traceback (most recent call last):
>   File "tensortrade/examples/ray_simple.py", line 31, in <module>
>     "env_config": {"corridor_length": 5}})
>   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/tune.py", line 543, in run
>     raise TuneError("Trials did not complete", incomplete_trials)
> ray.tune.error.TuneError: ('Trials did not complete', [PPO_SimpleCorridor_96575_00000])
> 

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by ijrsvt** [comment]
@cambel How did you build the Dockerfile exactly?

**Comment by cambel** [comment]
@ijrsvt I build the [Dockerfile](https://github.com/ray-project/ray/blob/master/docker/ray-ml/Dockerfile) using the requirements file under the [python](https://github.com/ray-project/ray/tree/master/python) directory (which could be made clearer in the README of the docker so that it is easier to use)
Other than that I just executed `docker build -t test .`

**Comment by ijrsvt** [comment]
@cambel That is a really good point, I can definitely improve the doc/usability of Docker images! I was wondering if there was a reason why our images on DockerHub are not sufficient? Here are the images for ([ray-ml](https://hub.docker.com/r/rayproject/ray-ml)). If you want to use those, you would want the images with the  `-gpu` tag (those have Nvidia Drivers). 


For building locally, you want to `docker build` with `--build-arg GPU=-gpu`. This ensures that your image is built off of our GPU images (not our CPU based images)!

**Comment by cambel** [comment]
@ijrsvt  Yes, my bad, I didn't notice there were images in DockerHub.

Before testing the DockerHub image I tried using `--build-arg GPU=-gpu` but It actually changes nothing since it is not used in the Dockerfile.

`[Warning] One or more build-args [GPU] were not consumed`

Using the docker image `rayproject/ray-ml:nightly-py36-gpu` I get the following error:

> ModuleNotFoundError: No module named 'ray.experimental.packaging'

when I run the followin script
``` python
import gym
from gym.spaces import Discrete, Box
from ray import tune

class SimpleCorridor(gym.Env):
    def __init__(self, config):
        self.end_pos = config["corridor_length"]
        self.cur_pos = 0
        self.action_space = Discrete(2)
        self.observation_space = Box(0.0, self.end_pos, shape=(1, ))

    def reset(self):
        self.cur_pos = 0
        return [self.cur_pos]

    def step(self, action):
        if action == 0 and self.cur_pos > 0:
            self.cur_pos -= 1
        elif action == 1:
            self.cur_pos += 1
        done = self.cur_pos >= self.end_pos
        return [self.cur_pos], 1 if done else 0, done, {}

tune.run(
    "PPO",
    config={
        "log_level": "DEBUG",
        "env": SimpleCorridor,
        "num_gpus": 1,
        "num_workers": 4,
        "env_config": {"corridor_length": 5}})
```

**Comment by ijrsvt** [comment]
Oh @cambel I think that is a transient failure with `nightly` (fixed by #14615)
